#248: Implemented redis availability check, installation and launch

ekondrashev opened the pull request:
#248: Implemented Redis availability check, installation and launch
* Aligned `install_db.sh` messaging

@emilianodellacasa Not quite if if there are any other dependencies to check?

0crat commented:
Job [#257](https://github.com/timelesslounge/timelessis/issues/257) is now in scope, role is `REV`

<!-- https://www.0crat.com/footprint/CF9LG1YCA/66085287-de25-4953-a0db-d5773339c303, version: 0.41.1, hash: ${buildNumber} -->

vladarefiev commented:
@ekondrashev Great, but a couple of questions.

0crat commented:
This pull request [#257](https://github.com/timelesslounge/timelessis/issues/257) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/9edbf453-54e3-41c2-a025-9ae9688e0f62); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

<!-- https://www.0crat.com/footprint/CF9LG1YCA/cf96fe5a-1d46-41d8-b7cf-1acd9f9dbe5a, version: 0.41.1, hash: ${buildNumber} -->

0crat commented:
Job `gh:timelesslounge/timelessis#257` already assigned to @vladarefiev, can't assign to @amihaiemil

<!-- https://www.0crat.com/footprint/CF9LG1YCA/6a05afb3-c9c5-4071-8986-d75bcf76671f, version: 0.41.1, hash: ${buildNumber} -->

0crat commented:
Job `gh:timelesslounge/timelessis#257` already assigned to @vladarefiev, can't assign to @amihaiemil

<!-- https://www.0crat.com/footprint/CF9LG1YCA/16ce9dd3-ee5f-485e-afe8-4a679670e21d, version: 0.41.1, hash: ${buildNumber} -->

ekondrashev commented on scripts/install/deploy/install_redis.sh:
```diff
@@ -1,8 +1,26 @@
+redis-cli ping
+if [ "$?" -gt "0" ]; then
+  echo "Redis Not running, launching"
+  src/redis-server > /dev/null &
```
@vladarefiev if it was installed, the binary was already copied to `/usr/local/bin/` according to the [doc](https://redis.io/topics/quickstart#installing-redis), so I'm not sure we need to `cd redis-stable`. But this means that `src/` should be removed since `redis-server` is expected to be accessible from the PATH, removing it

ekondrashev commented on scripts/install/deploy/install_redis.sh:
```diff
@@ -1,8 +1,26 @@
+if [ "$?" -gt "0" ]; then
+  echo "Redis Not installed, installing"
+  sudo apt install make gcc libc6-dev tcl
+  wget http://download.redis.io/redis-stable.tar.gz
```
@vladarefiev right, fixing. Also will add unarchived folder removal

ekondrashev commented on scripts/install/deploy/install_redis.sh:
```diff
@@ -1,8 +1,26 @@
+redis-cli ping
+if [ "$?" -gt "0" ]; then
+  echo "Redis Not running, launching"
+  src/redis-server > /dev/null &
```
@vladarefiev not sure, I'd add this as different puzzle to create a service file and start the service in case it is needed. @emilianodellacasa do we need to have a service for the redis-server so it will be started after restart of a host where the deployment is happening to?

emilianodellacasa replied:
@ekondrashev It would be better, thanks

emilianodellacasa replied:
@ekondrashev The same should be true for Postgres, after a restart of the server all should be up and running

ekondrashev commented:
@vladarefiev thank you, added an archive/unarchived folder removal after install and fixed the launching to not use `src` folder, since all the binaries were copied to `/usr/local/bin`.

Also responded with regard redis autostart.

vladarefiev commented:
@ekondrashev Thanks, good to me

vladarefiev commented:
@rultor merge

rultor commented:
> @rultor merge

@vladarefiev Thanks for your request. @emilianodellacasa Please confirm this.

emilianodellacasa commented:
@ekondrashev Just a single comment, please check

ekondrashev commented:
@emilianodellacasa sorry for the delay, missed the comment. Pushed 2 puzzles to create/install/start services for Redis and Postgres. Pls lmk if all is fine

ekondrashev commented:
@emilianodellacasa letm me fix travis issues

emilianodellacasa commented:
@rultor merge

rultor commented:
> @rultor merge

@emilianodellacasa OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/17872-465055550)

rultor commented:
> @rultor merge

@emilianodellacasa Done! FYI, the full log is [here](http://www.rultor.com/t/17872-465055550) (took me 7min)

0crat commented:
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

<!-- https://www.0crat.com/footprint/CF9LG1YCA/1a16c4da-3994-4329-a272-87a080103773, version: 0.41.15, hash: ${buildNumber} -->

0crat commented:
The job [#257](https://github.com/timelesslounge/timelessis/issues/257) is now out of scope

<!-- https://www.0crat.com/footprint/CF9LG1YCA/4d24baff-0e4b-49b6-84fd-ce397ff88f06, version: 0.41.15, hash: ${buildNumber} -->

0crat commented:
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

<!-- https://www.0crat.com/footprint/CF9LG1YCA/03226e20-6f3d-4f05-a9f7-00adc7ca19cc, version: 0.41.15, hash: ${buildNumber} -->

ypshenychka commented:
@vladarefiev According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews

vladarefiev commented:
@ypshenychka I confirm

ypshenychka commented:
@vladarefiev thanks

ypshenychka commented:
@0crat quality acceptable

0crat commented:
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

<!-- https://www.0crat.com/footprint/CF9LG1YCA/5d96c822-c44e-48b9-b427-d7dee7671134, version: 0.41.15, hash: ${buildNumber} -->

0crat commented:
Order was finished, quality is "acceptable": +15 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

<!-- https://www.0crat.com/footprint/CF9LG1YCA/b762b9a3-0bfb-43de-bee8-89ff70888299, version: 0.41.15, hash: ${buildNumber} -->
